flexinput/src/main/java/com/lytefast/flexinput/utils/ThumbnailUtils.kt

```diff
@@ -0,0 +1,25 @@
+package com.lytefast.flexinput.utils
+
+import android.content.ContentResolver
+import android.graphics.Bitmap
+import android.net.Uri
+import android.os.Build
+import android.util.Log
+import android.util.Size
+import androidx.annotation.RequiresApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+object ThumbnailUtils {
+
+  @RequiresApi(Build.VERSION_CODES.Q)
+  suspend fun getThumbnailQAsync(contentResolver: ContentResolver, uri: Uri, width: Int, height: Int): Bitmap? = withContext(Dispatchers.IO) {
```

Comment:
A future non-legacy implementation will expand on this class

Reply:
Can probably drop async as the `suspend` keyword already denotes that.

Reply:
na this was feedback from Andy I want to preserve, indicating this is an async method.

Reply:
agree to disagree. The compiler will complain if you try to do this sync, but w/e